### PR TITLE
Chiachun/pwx 36227 pool delete tests update

### PR DESCRIPTION
**Pull request by chiachunlin**
fix pool delete tests
add PoolDeleteMultiplePools
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: tests/basic/storage_pool_test.go [hunk 1]
```diff
@@ -10212,41 +10213,73 @@ func findNodeForReplAdd(vol *volume.Volume) (*node.Node, error) {
...
-		log.InfoD("cannot find nonkvdb node, select kvdb node %v for test", testNode.Addresses)
-		poolsMap, err := Inst().V.GetPoolDrives(testNode)
+
+	if len(testNodes) == 0 {
+		testNode := stNodes[0]
+		log.InfoD("select kvdb node %v for test, the node need to have at least 2 pools", testNode.Name)
+		poolsMap, err := Inst().V.GetPoolDrives(&testNode)
 		log.FailOnError(err, "cannot get pool drives")
-		log.InfoD("node %v has pools %+v", testNode.Addresses, poolsMap)
+		log.InfoD("the kvdb node %v has pools %+v", testNode.Name, poolsMap)
```
Comment (lsrinivas-pure) [inline tests/basic/storage_pool_test.go:10263]: how are you making sure that stNodes[0] is a kvdb node?

Comment (chiachunlin) [reply tests/basic/storage_pool_test.go:10263]: I update the comments

File: drivers/volume/portworx/portworx.go [hunk 2]
```diff
@@ -1203,6 +1203,14 @@ func (d *portworx) EnterPoolMaintenance(n node.Node) error {
 }
 
 func (d *portworx) ExitPoolMaintenance(n node.Node) error {
+
+	// no need to exit pool maintenance if node status is up
+	pxStatus, err := d.GetPxctlStatus(n)
+	if err == nil && pxStatus == api.Status_STATUS_OK.String(){
```
[the remark this answers is not in the document]
Comment (chiachunlin) [reply drivers/volume/portworx/portworx.go:1209]: I reused your function and this is the call stack. 
```
deletePoolAndValidate() storage_pool_test.go
  --> DeleteGivenPoolInNode()  common.go
     --> EnterPoolMaintenance() portworx.go
     --> ExitPoolMaintenance()  portworx.go 
```
for your suggestion, the call stack will be 
```
deletePoolAndValidate() storage_pool_test.go
  --> DeleteGivenPoolInNode()  common.go
     --> EnterPoolMaintenance() portworx.go
     --> exitPoolMaintenance()  storage_pool_test.go  <<<<<<
```
I think it's better to keep the original solution. It makes more sense. 
And current ExitPoolMaintenance() will fail if PX is not in maintenance mode. It happens while deleting the last pool, PX automatically enters Up mode (storageless node now) then the test will fail.

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Pull request merged**